internal/backend/remote-state/azure/backend_state.go

```diff
@@ -57,42 +51,30 @@ func (b *Backend) DeleteWorkspace(ctx context.Context, name string, _ bool) erro
 
 	ctx, cancel := b.getContextWithTimeout(ctx)
 	defer cancel()
-	client, err := b.armClient.getBlobClient(ctx)
-	if err != nil {
-		return err
-	}
+	blobClient := b.containerClient.NewBlockBlobClient(b.path(name))
 
-	if resp, err := client.Delete(ctx, b.armClient.storageAccountName, b.containerName, b.path(name), blobs.DeleteInput{}); err != nil {
-		if resp.Response.StatusCode != 404 {
-			return err
+	if _, err := blobClient.Delete(ctx, nil); err != nil {
+		if !notFoundError(err) {
+			return fmt.Errorf("error deleting blob: %w", err)
 		}
 	}
 
 	return nil
 }
 
 func (b *Backend) StateMgr(ctx context.Context, name string) (statemgr.Full, error) {
-	ctx, cancel := b.getContextWithTimeout(ctx)
-	defer cancel()
-
-	blobClient, err := b.armClient.getBlobClient(ctx)
-	if err != nil {
-		return nil, err
-	}
+	blobClient := b.containerClient.NewBlockBlobClient(b.path(name))
 
 	client := &RemoteClient{
-		giovanniBlobClient: *blobClient,
-		containerName:      b.containerName,
-		keyName:            b.path(name),
-		accountName:        b.accountName,
-		snapshot:           b.snapshot,
-		timeoutSeconds:     b.armClient.timeoutSeconds,
+		blobClient: blobClient,
+		snapshot:   b.snapshot,
+		timeout:    b.timeout,
 	}
 
 	stateMgr := remote.NewState(client, b.encryption)
 
 	// Grab the value
-	if err := stateMgr.RefreshState(context.TODO()); err != nil {
+	if err := stateMgr.RefreshState(ctx); err != nil {
```

Comment:
We are intentionally leaving these as context.TODO() for now. I hit the same thing a few weeks ago.

We want to make sure that the context used in the backends for critical operations can not be cancelled externally and have not determined the best way to accomplish that yet.